frontend/public/kubevirt/components/modals/restart-vm-modal.jsx

```diff
@@ -3,18 +3,31 @@ import PropTypes from 'prop-types';
 
 import { PromiseComponent } from '../utils/okdutils';
 import { createModalLauncher, ModalTitle, ModalBody, ModalSubmitFooter } from '../factory/okdfactory';
-import { k8sKill, k8sGet } from '../../module/okdk8s';
+import { k8sKill, k8sGet, k8sPatch } from '../../module/okdk8s';
 import { VirtualMachineInstanceModel } from '../../models';
+import { getPxeBootPatch } from 'kubevirt-web-ui-components';
 
 class RestartVmModal extends PromiseComponent {
   constructor(props) {
     super(props);
     this._cancel = this.props.cancel.bind(this);
     this._submit = this.submit.bind(this);
+    this._killVmi = this.killVmi.bind(this);
   }
 
   submit(event) {
     event.preventDefault();
+
+    const pxePatch = getPxeBootPatch(this.props.resource);
+    if (pxePatch.length > 0) {
+      const pxePromise = k8sPatch(this.props.kind, this.props.resource, pxePatch);
+      this.handlePromise(pxePromise).then(this._killVmi);
+    } else {
+      this._killVmi();
+    }
+  }
+
+  killVmi() {
```

Comment:
Dialog does not show error or do anything if we fail.
Also disabling the buttons could help when in progress.